src/coreComponents/dataRepository/DataContext.hpp

```diff
@@ -0,0 +1,233 @@
+  DataFileContext( xmlWrapper::xmlNode const & targetNode, xmlWrapper::xmlNodePos const & nodePos );
+  /**
+   * @brief Construct the file context of a Group from an xml node.
+   * @param targetNode the xml node containing the xml attribute
+   * @param att the target object xml attribute
+   * @param attPos the target object xml attribute position
+   */
+  DataFileContext( xmlWrapper::xmlNode const & targetNode, xmlWrapper::xmlAttribute const & att,
+                   xmlWrapper::xmlAttributePos const & attPos );
+
+  /**
+   * @return the target object name followed by the the file and line declaring it.
+   */
+  string toString() const override;
+
+  /**
+   * @return the type name in the source file (XML node tag name / attribute name).
+   */
+  string getTypeName() const
+  { return m_typeName; }
+
+  /**
+   * @return the source file path where the target object has been declared.
+   */
+  string getFilePath() const
+  { return m_filePath; }
+
+  /**
+   * @return the line (starting from 1) where the target object has been declared in the source file.
+   */
+  size_t getLine() const
+  { return m_line; }
+
+  /**
+   * @return the character offset in the line (starting from 1) where the target object has been
+   * declared in the source file.
+   */
+  size_t getOffsetInLine() const
+  { return m_offsetInLine; }
+
+  /**
+   * @return the character offset (starting from 0) in the source file path where the target object has been
+   * declared.
+   */
+  size_t getOffset() const
+  { return m_offset; }
+
+private:
+
+  /// @see getTypeName()
+  string const m_typeName;
+  /// @see getFilePath()
+  string const m_filePath;
+  /// @see getLine()
+  size_t const m_line;
+  /// @see getLineOffset()
+  size_t const m_offsetInLine;
+  /// @see getOffset()
+  size_t const m_offset;
+
+  /**
+   * @copydoc DataContext::getToStringInfo()
+   */
+  ToStringInfo getToStringInfo() const override;
+
+};
+
+
+} /* namespace dataRepository */
+} /* namespace geos */
+
+
+
+/// Formatter to be able to directly use a DataContext as a GEOS_FMT() argument.
+template<>
+struct GEOS_FMT_NS::formatter< geos::dataRepository::DataContext >
+{
+  /**
+   * @brief Format the specified DataContext to a string.
+   * @param dataContext the DataContext object to format
+   * @param ctx formatting state consisting of the formatting arguments and the output iterator
+   * @return iterator to the output buffer
+   */
+  auto format( geos::dataRepository::DataContext const & dataContext, format_context & ctx )
+  {
+    return format_to( ctx.out(), dataContext.toString() );
+  }
+
+  /**
+   * @brief Method to parse a dataContext from a string. Not implemented!
+   * @param ctx formatting state consisting of the formatting arguments and the output iterator
+   * @return iterator to the output buffer (leaved unchanged)
+   */
+  auto parse( format_parse_context & ctx )
+  {
+    GEOS_ERROR( "DataContext parsing is not implemented." );
+    return ctx.begin();
+  }
```

Comment:
`formatter.parse()` is not meant for parsing the target type itself, but for parsing the format string for the type and determining the format options, such as `{:<.2f}`. So anytime you do `GEOS_FMT("{}", dataContext)`, `parse` will be called and throw an error. 

To do a formatter correctly, you have 2 options:
* If you want to strictly only allow empty format strings `{}`, just returning `ctx.begin()` is sufficient - the library will throw an exception if the next character is not a `}`.
* Alternatively, you could inherit from `GEOS_FMT_NS::formatter<std::string>` and automatically reuse its `parse` method. This will give you free support for width/alignment specifiers. Then in format you can just
```c++
    return GEOS_FMT_NS::formatter<std::string>::format( dataContext.toString(), ctx );
```

Reply:
Ok! I will go with this second option.